pkg/fanal/image/registry/ecr/ecr.go

```diff
@@ -46,11 +47,34 @@ func (e *ECR) CheckOptions(domain string, option types.RegistryOptions) error {
 		return err
 	}
 
+	// override region with the value from the repository domain
+	cfg.Region = *region
+
 	svc := ecr.NewFromConfig(cfg)
 	e.Client = svc
 	return nil
 }
 
+// Endpoints take the form
+// <registry-id>.dkr.ecr.<region>.amazonaws.com
+// <registry-id>.dkr.ecr-fips.<region>.amazonaws.com
+// <registry-id>.dkr.ecr.<region>.amazonaws.com.cn
+// <registry-id>.dkr.ecr.<region>.sc2s.sgov.gov
+// <registry-id>.dkr.ecr.<region>.c2s.ic.gov
+// see
+// - https://docs.aws.amazon.com/general/latest/gr/ecr.html
+// - https://docs.amazonaws.cn/en_us/aws/latest/userguide/endpoints-arns.html
+// - /usr/local/aws-cli/awscli/botocore/data/endpoints.json
```

Comment:
Is https://github.com/boto/botocore/blob/1b6648509bead576aa0bbd433c57b32540c9313a/botocore/data/endpoints.json same file?

Reply:
It looks to be, and appears to be newer than the version I have installed in the aws cli.  I can change the comment to refer to this instead.

Reply:
Replaced with a link to the tagged file